### ISSUE TITLE
Unable to change the learning rate when restarting from a previous checkpoint.

### ISSUE THREAD

**Issue by arshadshk** [OpenNMT/OpenNMT-py#1911]
**Earlier I kept the learning_rate as 0.001 and am using adam optimizer.
I trained my model till 6000 steps.
Now I need to reduce the learning rate, so I changed the train.yaml file with learning_rate: 0.00001.
But when I restarted the training from checkpoint my_checpoint_6000.pt, in the printed logs the learning rate is still 0.001.**

( I don't want to initialize a lr_decay at the start, I want to change the learning rate on the go based on my intuition.)

Any guesses why this is happening?

**Comment by arshadshk** [comment]
One way is to read the checkpoint file using torch and edit the learning rate, and then use this checkpoint for the next training session.
Following is the code that works for such situations- 

```
import torch
t = torch.load('my_checpoint_6000.pt')
t['optim']['optimizer']['param_groups'][0]['lr'] = 0.00001
t['opt'].learning_rate = 0.00001
torch.save( t , 'my_checpoint_6000.pt')

```

( It seems that whenever a session is resumed from a past checkpoint, all the arguments parsed via train.yaml files are ignored, and those arguments are loaded from the ".pt" file itself. 
Still, I hope someone more familiar with this framework makes these changes. )

**Comment by francoishernandez** [comment]
You're looking for the `reset_optim`flag. #1393
